File: nom-api/Nom.Orch/Models/Measurement/MeasurementConversionModel.cs
namespace Nom.Orch.Models.Measurement
{
    /// <summary>
    /// Model representing a measurement conversion rule in the API layer.
    /// </summary>
    public class MeasurementConversionModel
    {
        public long Id { get; set; }
        public long FromMeasurementId { get; set; }
        public string FromMeasurementName { get; set; } = string.Empty;
        public string FromMeasurementSymbol { get; set; } = string.Empty;
        public long ToMeasurementId { get; set; }
        public string ToMeasurementName { get; set; } = string.Empty;
        public string ToMeasurementSymbol { get; set; } = string.Empty;
        public decimal ConversionFactor { get; set; }
        public decimal? Offset { get; set; }
        public string? Formula { get; set; }
        public bool IsDirectConversion { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime? LastModifiedDate { get; set; }
    }
}

File: nom-ui/src/app/measurement/models/measurement-conversion.model.ts
export interface MeasurementConversionModel {
    id: number;
    fromMeasurementId: number;
    fromMeasurementName: string;
    fromMeasurementSymbol: string;
    toMeasurementId: number;
    toMeasurementName: string;
    toMeasurementSymbol: string;
    conversionFactor: number;
    offset?: number;
    formula?: string;
    isDirectConversion: boolean;
    createdDate: Date;
    lastModifiedDate?: Date;
    authorId: number;
    createdById: number;
    userId: number;
}
